lib/datadog/appsec/component.rb

```diff
@@ -28,6 +29,22 @@ def build_appsec_component(settings, telemetry:)
 
         private
 
+        def enable_appsec?(settings)
+          return false unless settings.respond_to?(:appsec) && settings.appsec.enabled
+
+          ffi_version = Gem.loaded_specs['ffi'] && Gem.loaded_specs['ffi'].version
+          if RUBY_VERSION >= '3.3.0' && ffi_version < '1.16.0'
+            Datadog.logger.warn(
+              'Appsec is not supported in Ruby versions above 3.3.0 when using `ffi` versions older than 1.16.0, ' \
+              'and will be forcibly disabled. This is due to a memory leak in `ffi` versions lower than 1.16.0.' \
+              'Please upgrade your `ffi` version to 1.16.0 or higher.'
```

Comment:
WDYT if we join the reason to the main sentence?

```suggestion
              'AppSec is not supported in Ruby versions above 3.3.0 when using `ffi` versions older than 1.16.0, ' \
              'and will be forcibly disabled due to a memory leak in `ffi`. ' \
              'Please upgrade your `ffi` version to 1.16.0 or higher.'
```

Reply:
yes, this is definitely better, thanks!

Reply:
fixed: [b7ca70a](https://github.com/DataDog/dd-trace-rb/pull/3969/commits/b7ca70ab82e390bb76527f9356162930e706b830)